modules/dxp/apps/portal-search-tuning/portal-search-tuning-rankings-web/src/main/java/com/liferay/portal/search/tuning/rankings/web/internal/index/RankingIndexReindexer.java

```diff
@@ -37,83 +37,74 @@
-			List<Long> classPKs = jsonStorageEntryLocalService.getClassPKs(
-				companyId, classNameLocalService.getClassNameId(Ranking.class),
-				QueryUtil.ALL_POS, QueryUtil.ALL_POS);
-
-			RankingIndexName rankingIndexName =
-				rankingIndexNameBuilder.getRankingIndexName(companyId);
+		List<Long> classPKs = jsonStorageEntryLocalService.getClassPKs(
+			companyId, classNameLocalService.getClassNameId(Ranking.class),
+			QueryUtil.ALL_POS, QueryUtil.ALL_POS);
 
-			if (ListUtil.isEmpty(classPKs)) {
-				if (_log.isInfoEnabled()) {
-					_log.info(
-						StringBundler.concat(
-							"Not reindexing ", rankingIndexName.getIndexName(),
-							" because the database has no ranking entries"));
-				}
+		RankingIndexName rankingIndexName =
+			rankingIndexNameBuilder.getRankingIndexName(companyId);
 
-				continue;
+		if (ListUtil.isEmpty(classPKs)) {
+			if (_log.isInfoEnabled()) {
+				_log.info(
+					StringBundler.concat(
+						"Not reindexing ", rankingIndexName.getIndexName(),
+						" because the database has no ranking entries"));
 			}
 
-			Date date = null;
+			return;
+		}
+
+		Date date = null;
 
-			if (_isExecuteSyncReindex(executionMode)) {
-				date = new Date();
+		if (_isExecuteSyncReindex(executionMode)) {
+			date = new Date();
 
-				Thread.sleep(1000);
-			}
-			else {
-				if (_log.isInfoEnabled()) {
-					_log.info(
-						"Deleting index " + rankingIndexName.getIndexName());
-				}
-
-				try {
-					rankingIndexCreator.delete(rankingIndexName);
-				}
-				catch (RuntimeException runtimeException) {
-					_log.error(
-						"Unable to delete index " +
-							rankingIndexName.getIndexName(),
-						runtimeException);
-				}
+			Thread.sleep(1000);
+		}
+		else {
+			if (_log.isInfoEnabled()) {
+				_log.info("Deleting index " + rankingIndexName.getIndexName());
 			}
 
-			if (!_isExecuteSyncReindex(executionMode)) {
-				if (_log.isInfoEnabled()) {
-					_log.info(
-						"Creating index " + rankingIndexName.getIndexName());
-				}
-
-				rankingIndexCreator.create(rankingIndexName);
+			try {
+				rankingIndexCreator.delete(rankingIndexName);
 			}
+			catch (RuntimeException runtimeException) {
+				_log.error(
+					"Unable to delete index " + rankingIndexName.getIndexName(),
+					runtimeException);
+			}
+		}
 
-			for (long classPK : classPKs) {
-				rankingIndexWriter.create(
-					rankingIndexName, _buildRanking(classPK));
+		if (!_isExecuteSyncReindex(executionMode)) {
+			if (_log.isInfoEnabled()) {
+				_log.info("Creating index " + rankingIndexName.getIndexName());
 			}
 
-			if (_isExecuteSyncReindex(executionMode)) {
-				SyncReindexManager syncReindexManager =
-					_syncReindexManagerSnapshot.get();
+			rankingIndexCreator.create(rankingIndexName);
+		}
 
-				syncReindexManager.deleteStaleDocuments(
-					rankingIndexName.getIndexName(), date,
-					Collections.emptySet());
-			}
+		for (long classPK : classPKs) {
+			rankingIndexWriter.create(rankingIndexName, _buildRanking(classPK));
```

Comment:
should we be calling reindexStatusMessageSender.sendStatusMessage(String className, long count, long total) during this loop (maybe at some interval) to report progress on the "document" level as well as the company level?

Reply:
Actually, we can do that, but to do so we would have to add new logic and we would depend on each implementation of indexReindexer using reindexStatusMessageSender.sendStatusMessage() to report the status. Also, a quick test here to see if it would work, I could say that the back-end can track progress at the "document" level without problems, but the front-end is not showing progress in real time.

Reply:
So I just decided to use the actual logic and if we want to improve it later we can create a ticket